lib/Sema/ConstraintSystem.cpp

```diff
@@ -3711,24 +3712,27 @@ static bool diagnoseConflictingGenericArguments(ConstraintSystem &cs,
       genericParams;
   // Consider only representative type variables shared across
   // all of the solutions.
```

Comment:
I don't believe this comment is accurate anymore as the logic is no longer limited to type variables shared across every solution, e.g if a type variable is unique only to a couple of solutions, we'll now diagnose any binding differences. I think that's probably reasonable, as the condition above restricts this logic to only happen when overloads are the same across solutions. But we should update this comment to reflect that. Does that sound right @xedin?

If we did want to maintain the behavior of only considering type variables shared by every solution, we could simplify this loop slightly by iterating over only the first solution's type variables. And the check later should probably become:

```cxx
if (!type) {
  arguments.clear();
  break;
}
```

Reply:
For what I understand it makes sense to just update the comment. I can send a follow up today after work to fix that no problem!

Reply:
Yeah, I missed that comment, was thinking about whether `if (!type)` makes sense, it seems like we ought to diagnose  situations like that as well but I'm still torn on that...